### DISCUSSION TITLE
can priority field be negative? ie `priority: -1`

### DISCUSSION THREAD

**Discussion by tooptoop4** [argoproj/argo-workflows#11669] [Q&A] [answered]
so that 1 cronworkflow has less priority than all other workflows (which i assume default to 0)?

**Comment by agilgur5** [comment, marked as answer]
`priority` is an [`int32`](https://github.com/argoproj/argo-workflows/blob/f5e31f8f36b32883087f783cb1227490bbe36bbd/pkg/apis/workflow/v1alpha1/common.go#L73) all [the](https://github.com/argoproj/argo-workflows/blob/f5e31f8f36b32883087f783cb1227490bbe36bbd/pkg/apis/workflow/v1alpha1/workflow_types.go#L360) way [down](https://github.com/argoproj/argo-workflows/blob/f5e31f8f36b32883087f783cb1227490bbe36bbd/workflow/sync/throttler.go#L145C5-L145C5), so I do believe you can pass negatives to it. See also the [Field Reference](https://argoproj.github.io/argo-workflows/fields/#workflowspec) in the docs.

It does seem to [default to 0](https://github.com/argoproj/argo-workflows/blob/f5e31f8f36b32883087f783cb1227490bbe36bbd/workflow/sync/sync_manager.go#L159) as well (which is also the [zero value](https://go.dev/tour/basics/12) of ints in Go).
